### DISCUSSION TITLE
Support for SMIME

### DISCUSSION THREAD

**Discussion by 640kb** [cutelyst/simple-mail#100] [General]
Hello,

would you be interested in adding support for signing and/or encrypting emails?

I know the topic is not that easy and I would offer my help. I am currently working on it and it could be implemented in simpleMail with the help of OpenSSL.

**Comment by dantti** [comment]
As long as it's optional I PR are welcome :)

**Comment by 640kb** [comment]
Good to hear!

With my current changes you can already sign and encrypt. But it still lacks support for file attachments, for example, and at the moment I've only implemented it for PKCS12 certificates.

I'm still in the process of improving and simplifying the code.
I would hopefully submit a pull request soon and then we can discuss the changes and see what can perhaps be done better.

The changes are currently based on the 2.x.x branch, but I think the necessary adjustments for the master branch should not be too difficult.
